cmd/cluster/aws/create.go

```diff
@@ -77,10 +85,89 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 	if err != nil {
 		return err
 	}
-	infraID := opts.InfraID
-
 	// Load or create infrastructure for the cluster
 	var infra *awsinfra.CreateInfraOutput
+	var iamInfo *awsinfra.CreateIAMOutput
+	infraID := opts.InfraID
+
+	if infraID == "" {
+		return ErrMsg("infra-id")
+	}
+	hci := hyperv1.HostedClusterInfrastructure{
```

Comment:
Can all this live in a function getInfraFromHCI?

Reply:
done.